### PR TITLE
add /MT for MSVC static lib

### PR DESCRIPTION
**Pull request by AlexP11223**
Added /MT flag for MSVC static build (`cmake -DG3_SHARED_LIB=OFF`), as mentioned in #275

Without /MT it results in errors like this:
```
1>g3logger.lib(logcapture.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(loglevels.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(g3log.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(logworker.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(crashhandler_windows.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(logmessage.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(filesink.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(stacktrace_windows.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>g3logger.lib(time.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in ConsoleApplication9.obj
1>MSVCRT.lib(MSVCR120.dll) : error LNK2005: _memmove already defined in LIBCMT.lib(memmove.obj)
1>MSVCRT.lib(MSVCR120.dll) : error LNK2005: _memchr already defined in LIBCMT.lib(memchr.obj)
```

### CODE REVIEW

File: Build.cmake [hunk 1]
```diff
@@ -116,6 +116,10 @@ ENDIF()
       ENDIF()
       ADD_LIBRARY(${G3LOG_LIBRARY} SHARED ${SRC_FILES})
    ELSE()
+      IF (MSVC)
+          set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
```
Comment (KjellKod) [inline Build.cmake]: Looks good. Ref: https://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx

**Comment by KjellKod** [comment]
👍

**Comment by englercj** [comment]
Being a static library doesn't imply `/MT`. That is only if you statically link the *runtime library*. This needs to be an option as it will break all my apps I use g3log in because they are all `/MD` (which is the default).

It is generally not recommended to use `/MT` because you will bake a version of the CRT into you app, which means you will have a bad time when the system upgrades and you have to recompile. There is a fairly narrow set of circumstances where you actually want `/MT`.

**Comment by KjellKod** [comment]
@englercj thanks for the quick feedback. 
@AlexandrePTJ please address this with CMake options so it doesn’t break for other users 
I will revert the changes in: https://github.com/KjellKod/g3log/pull/279

**Comment by KjellKod** [comment]
This PR is reverted in https://github.com/KjellKod/g3log/pull/279

**Pull request merged**